test/functional/cmdline_options_tester/src/Tokenizer.java

```diff
@@ -81,30 +81,31 @@ private String token() throws Exception {
          * otherwise it will ends up returning an empty argument to be ignored in forming the command array.
          *
          * Three cases are taken into account in the while loop:
-         * 1) Strings in double-quotes (a argument starting with double-quotes(")) are completely extracted from the command string,
-         *    whether white spaces in the argument exist or not. Double-quotes themselves are discarded after extraction when scanning forward.
-         * 2) Normal arguments (no double-quotes occur in the argument) are split up and extracted with white spaces between arguments.
+         * 1) Strings in double-quotes/single-quotes (an argument starting with double-quotes(") or single-quotes(')) are completely extracted from the command string,
+         *    whether white spaces in the argument exist or not. Double-quotes/single-quotes themselves are discarded after extraction when scanning forward.
+         * 2) Normal arguments (no double-quotes/single-quotes occur in the argument) are split up and extracted with white spaces between arguments.
          * 3) Following the original order of the command line, non-quoted portions of an argument would be extracted in case 2)
          *    and concatenated to quoted portions of the argument extracted in case 1) within the external while loop.
          */
         while ((index < buffer.length()) && !Character.isWhitespace(buffer.charAt(index))) {
 
         	/**
-        	 * Case 1: Expect a double-quoted string once a double-quotes is detected
+        	 * Case 1: Expect a double-quoted or single-quoted string once either is detected
         	 */
-            if (buffer.charAt(index) == '"') {
-                index++;
-                int i = index;
+					  char delim = buffer.charAt(index);
+            if ((delim == '"') || (delim == '\'')) {
+								index++;
+								int i = index;
 
                 /**
-                 * search for another double-quotes since double-quotes appear in pairs.
+                 * search for another double-quotes/single-quotes since they appear in pairs.
```

Comment:
Please fix the grammar here (we're searching for a singular quote).

Reply:
do you mean like this: "... another double-quotes/single-__quote__ since they..."?

Reply:
If I were writing that comment, I would say 'search for the closing quote' (I don't think we need to repeatedly say 'double-quote/single-quote').